src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -504,6 +504,19 @@ private sealed class FileSystemEnumerable<T> : IEnumerable<T>
                         }
                         searchPattern = searchPattern.Substring(lastSlash + 1);
                     }
+
+                    // Typically we shouldn't see either of these cases, an upfront check is much faster
+                    for (int i = 0; i < searchPattern.Length; i++)
```

Comment:
Nit: Could be simplified to `foreach (char c in searchPattern)`. Also, not surrounding the loop body with `{` and `}` looks out of place here to me -- it doesn't seem to fit in with the style of the surrounding code.